ADR/0055-revised-component-model.md

```diff
@@ -0,0 +1,628 @@
+  * Type: string
+  * Updated only during onboarding explicit/implicit
+  * If containerImage is modified after onboarding, just do nothing, not even add anything in status.message like we do now
+
+* **status.versions** (array)
+  * Type: array of objects
+  * All versions which were processed by onboarding (implicit/explicit)
+  * If version is removed from the spec, offboarding will remove it from the status
+
+* **status.versions[].name** (string)
+  * Type: string
+
+* **status.versions[].onboarding-status** (string)
+  * Type: string
+  * Onboarding status will be either succeeded or failed (disable won't be there because we will just remove specific version section)
+
+* **status.versions[].configuration-merge-url** (string)
+  * Type: string
+  * Only present if onboarding was successful
+
+* **status.versions[].onboarding-time** (string)
+  * Type: string
+  * Only present if onboarding was successful
+
+* **status.versions[].revision** (string)
+  * Type: string
+
+* **status.versions[].skip-builds** (boolean)
+  * Type: boolean
+  * Updated when skip-builds changes in the spec for the version
+
+* **status.versions[].build-nudged-by** (array)
+  * Type: array of objects
+  * Set when other version/component specifies this version in build-nudges-ref
+
+* **status.versions[].build-nudged-by[].component** (string)
+  * Type: string
+
+* **status.versions[].build-nudged-by[].version** (string)
+  * Type: string
+
+* **status.versions[].build-nudges-ref** (array)
+  * Type: array of objects
+  * Updated when build-nudges-ref changes in the spec for the version
+
+* **status.versions[].build-nudges-ref[].component** (string)
+  * Type: string
+  * Mandatory
+
+* **status.versions[].build-nudges-ref[].version** (string)
+  * Type: string
+  * Mandatory
+
+* **status.versions[].message** (string)
+  * Type: string
+  * Version specific error message
+
+## Consequences
+
+* Improved Reusability: Component branches can be easily reused across multiple
+  component groups and release plans.
+* Simplified Model: The new structure is more flexible and intuitive, reducing
+  complexity for users.
+* Clear Ownership: The Group and Component resources will be clearly associated
+  by their controllers (integration-service and build-service respectively).
+* Migration Effort: A significant effort is required to design, implement, and
+  migrate all existing users and components to the new model.
+* UI/UX Changes: The user interface and experience will need to be completely
+  updated to reflect the new component group model.
+* Documentation Updates: All existing documentation related to components and
+  applications must be revised to reflect the new concepts and workflows.
+* Partial decoupling: This should give us some progress towards decoupling, but
+  the Snapshot resource will remain a common resource referenced by both
+  integration service and release service.
+* Nudging: The "nudging" feature will need design changes. There is some idea
```

Comment:
Is there a good reason to keep nudging in the build-service? I think it will fit mintmaker better since it's responsible for dependency updates.